packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2896,3 +2947,205 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+///
+/// The [selectionDetails] object provides information about the active
+/// selection contained under the [SelectionListener] this callback is provided to.
+typedef SelectionListenerSelectionChangedCallback = void Function(SelectionDetails selectionDetails);
+
+/// A [SelectionContainer] that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in this widgets subtree.
+///
+/// This widget does not listen to selection changes of nested [SelectionArea]s
+/// or [SelectableRegion]s in its subtree because those widgets create their own
+/// [SelectionRegistrar]s that do not register to any ancestor registrar.
```

Comment:
I think the doc here is reaching implementation detail. just say because the SelectableRegion is self-contain and do not bubble up selection.

Although I think at some point we want to support nested SelectableRegion, the selection start outside the SelectableRegion can still select into the SelectableRegion, but the selection starts inside the SelectableRegion can not move outside of it. Have you thought about how selectionListener should behave in this scenario?

Reply:
In that scenario, I think we can just treat the nested region as a `SelectionContainer` (it is already) and process its selection like any other child selectable. Does this sound reasonable?

Reply:
yes that was my original plan if we were to start nest selectableregion